test/Microsoft.AspNet.Mvc.IntegrationTests/DictionaryModelBinderIntegrationTest.cs

```diff
@@ -134,6 +137,7 @@ public class DictionaryModelBinderIntegrationTest
         [Theory]
         [InlineData("?[key0]=10")]
         [InlineData("?[0].Key=key0&[0].Value=10")]
+        [InlineData("?index=low&[low].Key=key0&[low].Value=10")]
```

Comment:
Do we need to add integration tests for these cases? Are there test cases using this key-syntax that trigger validation errors?